jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -257,21 +266,19 @@ object XGBoost extends Serializable {
    * @param useExternalMemory indicate whether to use external memory cache, by setting this flag as
    *                          true, the user may save the RAM cost for running XGBoost within Spark
    * @param missing the value represented the missing value in the dataset
-   * @param baseMargin initial prediction for boosting.
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
   @throws(classOf[XGBoostError])
   def trainWithRDD(
-      trainingData: RDD[MLLabeledPoint],
+      trainingData: RDD[XGBLabeledPoint],
```

Comment:
I agree that we should not allow wrong results and surely exceptions shouldn't not be silent. I look at the issue from a different angle. Let's say we've designed the API on top of `zip` and propagated its assumptions to `trainWithRDD`. The users called it with their data and ends up with an exception similar to the one above. I claim that there is not much one can do to get out of this situation. The only way I've managed to find is

```
val training = ...
val trainingPoints = training.map(_.point)
val baseMargin = training.map(_.baseMargin)
```

which is essentially what this PR does. Of course, we can leave everything as-is and make the user responsible for writing this boilerplate, or even provide ready-made snippets to copy/paste. However, I feel that a more user-friendly approach in the long term is to change the API in such a way that it is harder to get wrong, even though short-term it might be uncomfortable for the users [*].

[*]: Note that this is not necessarily the case, as discussed above in (1).